lib/src/metrics/source_lines_of_code/source_lines_of_code_metric.dart

```diff
@@ -73,11 +73,11 @@ class SourceLinesOfCodeMetric extends FunctionMetric<int> {
   Iterable<ContextMessage> _context(
     Declaration node,
     Iterable<int> linesWithCode,
-    ResolvedUnitResult source,
+    InternalResolvedUnitResult source,
   ) =>
       linesWithCode.map((lineIndex) {
         final lineStartLocation = SourceLocation(
-          source.unit?.lineInfo?.getOffsetOfLine(lineIndex) ?? 0,
+          source.unit.lineInfo?.getOffsetOfLine(lineIndex) ?? 0,
```

Comment:
@incendial I think we can take `lineInfo` directly from `source`, need to check all code base to find similar situations.

Reply:
Done, I've also added checks for `ResultState` to prevent possible exceptions when `lineInfo` is accessed directly.